Std/Data/HashMap/Lemmas.lean

```diff
@@ -4,12 +4,21 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Scott Morrison
 -/
 import Std.Data.HashMap.Basic
+import Std.Data.Array.Lemmas
 import Std.Util.ProofWanted
 
 namespace Std.HashMap
 
-@[simp] proof_wanted empty_find? [BEq α] [Hashable α] {a : α} :
-    (∅ : HashMap α β).find? a = none
+namespace Imp
+
+@[simp] theorem empty_buckets : (empty : Imp α β).buckets = ⟨mkArray 8 AssocList.nil, by simp⟩ := rfl
+
+@[simp] theorem empty_val [BEq α] [Hashable α] : (∅ : HashMap α β).val = Imp.empty := rfl
+
+end Imp
+
+@[simp] theorem empty_find? [BEq α] [Hashable α] {a : α} :
+    (∅ : HashMap α β).find? a = none := by simp [find?, Imp.find?, Imp.mkIdx, USize.modn_lt]
```

Comment:
```suggestion
    (∅ : HashMap α β).find? a = none := by simp [find?, Imp.find?]
```
The goal is closed without those two lemmas on my machine

Reply:
Fixed, this was due to my previous definition of `getElem_mkArray`